lib/cib/cib_file.c

```diff
@@ -869,29 +869,39 @@ cib_file_perform_op_delegate(cib_t * cib, const char *op, const char *host, cons
         free_xml(result_cib);
 
     } else if (query == FALSE) {
-        pcmk__xml_log_patchset(LOG_DEBUG, cib_diff);
+        pcmk__output_t *out = NULL;
+
+        rc = pcmk_rc2legacy(pcmk__log_output_new(&out));
+        CRM_CHECK(rc == pcmk_ok, goto done);
+
+        pcmk__output_set_log_level(out, LOG_DEBUG);
+        rc = out->message(out, "xml-patchset", cib_diff);
+        out->finish(out, pcmk_rc2exitc(rc), true, NULL);
+        pcmk__output_free(out);
+        rc = pcmk_ok;
+
         free_xml(in_mem_cib);
         in_mem_cib = result_cib;
         cib_set_file_flags(private, cib_file_flag_dirty);
     }
 
-    free_xml(cib_diff);
-
     if (cib->op_callback != NULL) {
         cib->op_callback(NULL, cib->call_id, rc, output);
     }
 
-    if (output_data && output) {
-        if(output == in_mem_cib) {
-            *output_data = copy_xml(output);
-        } else {
-            *output_data = output;
-        }
+    if ((output_data != NULL) && (output != NULL)) {
+        *output_data = (output == in_mem_cib)? copy_xml(output) : output;
+    }
 
-    } else if(output != in_mem_cib) {
+done:
```

Comment:
Are you going to need this label later, or is it a stray?

Reply:
I'm using it now, in the `else if (query == FALSE)` block. That way I can break out, free variables, and return if there's an output initialization problem.